src/action.rs

```diff
@@ -1,7 +1,35 @@
-//! The different actions that can be done.
+//! The different actions that can be executed via any given key.
 
 use crate::key_code::KeyCode;
 
+/// The different types of actions we support for key macros
+#[non_exhaustive] // Definitely NOT exhaustive!  Let's add more! Mouse events maybe? :)
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum SequenceEvent {
+    /// A keypress/keydown
+    Press(KeyCode),
+    /// Key release/keyup
+    Release(KeyCode),
+    /// For sequences that need to wait a bit before continuing
+    Delay {
+        /// A delay (in ticks) to wait before executing the next SequenceEvent
+        since: u32,
+        /// Number of ticks to wait before removing the Delay
+        ticks: u32,
```

Comment:
`since` gets updated with each regular call to `Layout.tick()`; it's the timekeeper (like everything else that uses `since`).  `ticks` tells us how long the `SequenceEvent::Delay` should last.  As soon as `since >= ticks` we know the `Delay` is done and can allow it to fall out of the `sequenced` queue.

The two values keep getting pulled and pushed back into `sequenced` until the `Delay` is done which is why they go together like that.

Reply:
Just have a `duration: u16` field, and decrement it until 0.